src/LinAlg/hiopMatrixRajaDense.cpp

```diff
@@ -1376,6 +1376,70 @@ double hiopMatrixRajaDense::max_abs_value()
   return maxv;
 }
 
+/**
+ * @brief Returns the value of the element with the maximum absolute value.
+ * 
+ * @todo Consider using BLAS call (<D>LANGE)
+ */
+void hiopMatrixRajaDense::row_max_abs_value(hiopVector &ret_vec)
+{  
+  assert(ret_vec.get_size() == m());
+  ret_vec.setToZero();
+
+  auto& vec = dynamic_cast<hiopVectorRajaPar&>(ret_vec);
+  vec.copyFromDev();
+  double* vd = vec.local_data_host();
+  
+  double* data = data_dev_;
+  
+  for (int irow = 0; irow < m_local_; irow++)
+  {
+    RAJA::ReduceMax<hiop_raja_reduce, double> norm(0.0);
+    RAJA::View<const double, RAJA::Layout<2>> Mview(data, m_local_, n_local_);
+    RAJA::forall<hiop_raja_exec>(RAJA::RangeSegment(0, n_local_),
+      RAJA_LAMBDA(RAJA::Index_type j)
+      {
+        norm.max(fabs(Mview(irow,j)));
+      });  
```

Comment:
Perhaps consider parallelizing the outer loop?

Reply:
yes, like the one for sparse matrix, parallelize over rows. I think it can be done without using reduce

Reply:
Correct, the reduce functionality can be implemented sequentially in that context. Parallelizing over matrix rows seems to be a good first step. We can think about optimizing this further later on.

Reply:
I tried to follow the suggestion given in [RAJA issue 246 ](https://github.com/LLNL/RAJA/issues/246), to implement an array of RAJA reducer. It turns out Will's solution only works on CPU.
Will move back to parallelizing over rows.

Reply:
Perhaps in this case a better reference would be RAJA tutorial on [matrix-matrix multiplication](https://raja.readthedocs.io/en/v0.13.0/tutorial/matrix_multiply.html).